### PR TITLE
Refactored YouTube search to optimize API calls

### PR DESCRIPTION
**Pull request by ramsvidor**
This refactory aims to improve how data is consumed from the YouTube Data API.

- Changed the way videos are searched to optimize YouTube Data API calls. Now, only a single global search is ran per each search criteria configured.
- Added new application configuration parameter - `published-within-hours` - which will be used to search videos just from the last hours configured with it.

### CODE REVIEW

File: modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/daemons/fetcher/YouTubeFetcher.scala [hunk 1]
```diff
@@ -139,51 +164,27 @@ object YouTubeFetcher {
...
+
+        updates <- searchInformation.traverse { searchInfo =>
+          for {
+            globalSearchResult <- youtubeFetcher.searchVideos(
+              s"constellation-${searchInfo.text}",
+              searchInfo.maxPerDay,
+              Some(LocalDateTime.now().minusHours(searchInfo.publishedWithinHours))
+            )
+            dataUpdates <- youtubeFetcher.fetchVideoUpdates(filteredLatticeUsers, searchInfo, globalSearchResult)
+            newUpdates <- youtubeFetcher.fetchVideoUpdates(newlyLoadedLatticeUsers, searchInfo, globalSearchResult)
```
Comment (IPadawans) [inline modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/daemons/fetcher/YouTubeFetcher.scala:183]: Sorry, but why are we fetching twice here? I mean filteredLatticeUsers and newlyLoadedLatticeUsers

**Review by IPadawans — APPROVED** [review]
Overall looks good, I just have a small question that I've written in the comments

**Review by IPadawans — CHANGES_REQUESTED** [review]
Ooh @ramsvidor I forgot to mention, try to squash your commits into a single one, we use this pattern in the company, so let's try to keep in the metagraphs

File: modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/daemons/fetcher/YouTubeFetcher.scala (hunk 1 above)
Comment (ramsvidor) [reply modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/daemons/fetcher/YouTubeFetcher.scala]: You're right. It doesn't make sense any more. I removed the duplicate call.

**Review by IPadawans — APPROVED** [review]

**Pull request merged**